rental/views.py

```diff
@@ -94,27 +112,69 @@ def hardware_location(request):
 
   return render(request,'hardware_location.html', {'inventories': inventories, 'free_inventories': free_inventories})
 
-def rent_device(request):
+def reserve_device(request):
   user = User.objects.get(username=request.GET['user'])
   event = Event.objects.get(name=request.GET['event'])
   device = Device.objects.get(name=request.GET['device'])
-  device_rentals = Rental.objects.filter(inventory__device=device, returned = False)
-  free_inventory = Inventory.objects.filter(Q(event=event,device=device) & ~Q(rental=device_rentals)) 
+  rentals = Rental.objects.filter(event = event, device = device, returned = False, reservation = False)
+  reservations = Rental.objects.filter(event = event, device = device, returned = False, reservation = True)
+  free_inventory = Inventory.objects.filter(event=event, device=device).count() - rentals.count() - reservations.count()
   response = {}
   response['username'] = user.username
 
   if user.socialaccount_set.count() > 0:
     response['avatar'] = user.socialaccount_set.all()[0].get_avatar_url()
   
-  if free_inventory.count() > 0:
-    new_rental = Rental(user = user, inventory = free_inventory[0])
+  if free_inventory > 0:
+    new_rental = Rental(user = user, event = event, device = device)
     new_rental.save()
-    rentals = Rental.objects.filter(inventory__event = event, inventory__device = device, returned = False)
-    free = free_inventory.count()
+    free = free_inventory
     return render(request, 'partials/device_rentals.html',   {'rentals':rentals, 'free': free})
```

Comment:
This should be fixed now. 
`rentals = Rental.objects.filter(inventory__event = event, inventory__device = device, returned = False)` wont work because `inventory` is no longer allocated upon creating a `rental` object.